app/controllers/application_controller.rb

```diff
@@ -1,5 +1,12 @@
 class ApplicationController < ActionController::Base
-  def current_user
-    @user = User.first
+  protect_from_forgery with: :exception
+
+  before_action :configure_permitted_parameters, if: :devise_controller?
+
+  protected
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up, keys: %i[name photo bio])
+    devise_parameter_sanitizer.permit(:account_update, keys: %i[name photo])
   end
 end
```

Comment:
**OPTIONAL** - You did well so far, but I noticed that you have only used the `before_action :authenticate_user!` module in your user controller, which means I can access the controller actions in the `post` controller without authentication. Please ensure that all controller action requires authentication before they can be accessed. Please fix this. I suggest adding the code below to your application_controller.

`before_action :authenticate_user!`